src/Sylius/Component/Core/Model/OrderItem.php

```diff
@@ -27,8 +27,46 @@ class OrderItem extends BaseOrderItem implements OrderItemInterface
     protected $variant;
 
     /**
+     * @var string
+     */
+    protected $productName;
+
+    /**
+     * @var string
+     */
+    protected $variantName;
+
+    /**
      * {@inheritdoc}
      */
+    public function getVariant(): ?ProductVariantInterface
+    {
+        return $this->variant;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setVariant(?ProductVariantInterface $variant): void
```

Comment:
Hmm, what if a product name changes after the product was added to the cart but before the checkout was completed?

Reply:
Good point 👍  What is your suggestion? Should I set immutable names on checkout `complete` transition?

Reply:
I've created `OrderImmutableNamesSetter` which is called on checkout `complete` transition.